File: Lab_3/src/Lab3/SmartBullet.java
package Lab3;
interface CaptureTarget {

	public String GetCapturedTaget();
	public String SetCapturedTarget(String Target);
}

public class SmartBullet extends Bullet implements CaptureTarget{

	private String Target;
	public SmartBullet(String name) {
		super(name);
		System.out.println("This is SmartBullet");
	}

	public String GetCapturedTaget() {
		return this.Target;
	}
	
	public String SetCapturedTarget(String Target) {
		this.Target = Target;
		return this.Target;
	}




}

File: Lab9/SmartBullet.py
from enum import Enum
from Bullet import Bullet

class SmartBullet(Bullet):
  def __init__(self, name: str, size: float, intact: bool, target: str):
    super().__init__(name, size, intact)
    self.__target = target

  def getTarget(self): return self.__target
  def setTarget(self, target: str): self.__target = target
